### PR TITLE
machine/badger2040: support for Badger 2040

### PR DESCRIPTION
**Pull request by deadprogram**
This PR provides support for Badger 2040.

### CODE REVIEW

**Comment by deadprogram** [comment]
@conejoninja it seems this is working now for you as expected? I changed to "ready for review".

File: src/machine/board_badger2040.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	I2C0_SDA_PIN Pin = GPIO4
+	I2C0_SCL_PIN Pin = GPIO5
+
+	I2C1_SDA_PIN Pin = NoPin
+	I2C1_SCL_PIN Pin = NoPin
+)
+
+// SPI pins.
+const (
+	SPI0_SCK_PIN Pin = NoPin
```
Comment (conejoninja) [inline src/machine/board_badger2040.go]: should we define SPI0_SCK_PIN as EPD_SCK_PIN, same for EPD_SDO_PIN ?

**Comment by conejoninja** [comment]
It's working fine. We still miss a few things:

- Exposed pins on the back (left) : 3v, sda, scl, int, tx, rx, swc, swd
- QSPI pins for flash access
- USB-CDC :(


*Neither buttons nor I²C were tested yet.

File: src/machine/board_badger2040.go (hunk 1 above)
Comment (deadprogram) [reply src/machine/board_badger2040.go]: Done.

**Comment by deadprogram** [comment]
Thanks for the review @conejoninja now merging.

**Pull request merged**